BlueBrain/Search
# MySQL support for `bbs_database` + integration test

Closes #431 

# What was done
* Made sure `bbs_database init` and `bbs_database add` support `--db-type=mysql`
* Introduced 2 new subfolders inside of `tests` (inspired by https://realpython.com/python-testing/#writing-integration-tests)
  * `unit`
  * `integration`
* Wrote `tests/integration/test_bbs_database.py` integration test that tests all the existing commands (`init`, `parse` and `add`) without any mocking/patching and using both `sqlite` and `mysql` backends

# How to test
```bash
pytest tests/integration/test_bbs_database.py
```
This should run both the `sqlite` and `mysql` test. If the `mysql` test is skipped then you probably don't have docker running on your computer (check `docker ps`). Also note that you need to get the MySQL image `docker pull mysql`.

# More details + to discuss
### So the `bbs_database` unit tests are only run for the `sqlite` backend, right?
Yes. This is purely because of the fact that the mysql docker container just takes too long to start. We have this `sqlite` and `mysql` logic in the current codebase (for the CORD19 database) and a lot of us just skip the `mysql` by doing `pytest -k "not mysql"`.
### Should we modify `tox.ini` and have different handling of unit and integrations tests?
I guess this is something we should decide on together. In general, the integration tests should not do any patching/mocking and are gonna take longer to run.

### Isn't the `docker` +`mysql` logic just living at two places now (`tests/conftest.py` and `tests/integration/test_bbs_database.py`)?
Yes, and there is a big overlap. However, I propose we create a separate issue for refactoring things. For now, our test suit is going to create 2 docker containers.
